src/database/type-transforms.tsx

```diff
@@ -109,3 +111,67 @@ export const buildUser = (document: firebase.firestore.DocumentData): User => {
 
   return { ...pick(data, ["name", "email"]), id: document.id };
 };
+
+const buildCovidCase = (
+  doc: firebase.firestore.DocumentData,
+): ShadowCovidCase => {
+  const data = doc.data();
+
+  const observedAt = parse(doc.id, "yyyy-MM-dd", startOfToday());
```

Comment:
Remind me why observedAt is start of today here?

Reply:
the API for this function is a little weird; in this case we are getting the entire date (year, month, day) from the parsed string and `startOfDay` is a "reference date" that is required to fill in any missing pieces of a full Date object that aren't parsed; in this case we are only using the time portion and the local time zone of "today", which has the effect of normalizing `observedAt` to the start of whatever day it is.

This addresses your Extreme Paranoia as well, since the change of date as you cross the midnight boundary will just be ignored. 

I will grant you that this is rather non-obvious; perhaps an explanatory comment or even a refactor into a function with a more descriptive name would be in order here.